R/generic_import.R

```diff
@@ -136,20 +139,23 @@ Import <- R6::R6Class(
       private$con <- db_connect(private$db_name, self$path)
 
       private$log_table <- db_get_log_table(private$db_name, self$path)
-      private$log <- ImportLog$new(private$con, private$log_table, self$path)
+      private$log <- ImportLog$new(private$con, private$log_table, self$path,
+                                   private$language, private$mode)
       private$require_branch <-
         private$repo_config$db[[private$db_name]]$require_branch
 
-      private$mode <- private$import_config$dettl$mode
-
       private$confirm <- private$repo_config$db[[private$db_name]]$confirm
     },
 
+    # nocov start
+    ## Safe to ignore this in coverage as it doesn't and won't ever be run.
+    ## Viewing this as abstract impl of functions which a subclass must define
     #' @description
     #' Abstract impl should be overridden by subclass
     read_config = function() {
       NULL
```

Comment:
if you just drop the `NULL` here, then none of the nocov faff is needed

Reply:
I tried that - it then causes an error in R6. Returns "attempt to apply non-function" e.g. when calling `import$load()`, even though the value of `load` has been overridden as a function. R6 treats methods and fields separately. It merges methods on class being instantaited & inherited ones, and the same for fields. Then adds methods to an env and then the fields to an env. Meaning if super class defines a field call `read_config` and subclass defines a method called `read_config` the field version will be available. See https://github.com/r-lib/R6/blob/5cd8acd0bc82488650460cbde30596052442d716/R/new.R#L116

Reply:
I was thinking that `read_config = function() {}` would work here?

Reply:
:facepalm: yes sorry misunderstood first comment. That is nicer, made the switch and removed nocov stuff.